daphne/src/roles.rs

```diff
@@ -321,7 +326,10 @@ pub trait DapLeader<'a, S>: DapAuthorizedSender<S> + DapAggregator<'a, S> {
             .await?;
         let reports = reports
             .into_iter()
-            .filter(|report| early_rejects.get(&report.metadata.id).is_none())
+            .filter(|report| {
+                task_config.expiration >= report.metadata.time
+                    || early_rejects.get(&report.metadata.id).is_none()
```

Comment:
We want the reports that are not expired AND not early rejects, but we have "||" here instead of "&&".  We should add unit test coverage for this filtering too.

Reply:
Great catch. I checked to make sure the Helper enforces this correctly. In fact, we already have a test to verify the Helper (see http_post_aggregate_init_expired_task() in daphne/src/roles_test.rs). We don't have a corresponding test for the Leader because the test framework for this crate (daphne/src/testing.rs) currently doesn't allow us to run run_agg_job() directly. (Basically we need a way to mock HTTP requests made by the Leader.)  It is feasible to add such a test to daphne_worker_test/tests/e2e.rs. I think we should add this, but I'd like to do so in a separate PR, as there is some e2e.rs refactoring that happens higher in the PR stack.

Reply:
Noted in issue https://github.com/cloudflare/daphne/issues/100.